datanode/api/trading_data_v2.go

```diff
@@ -3048,6 +3055,78 @@ func (t *tradingDataServiceV2) GetVegaTime(ctx context.Context, _ *v2.GetVegaTim
 
 // -- NetworkHistory --.
 
+func (t *tradingDataServiceV2) ExportNetworkHistory(req *v2.ExportNetworkHistoryRequest, stream v2.TradingDataService_ExportNetworkHistoryServer) error {
+	defer metrics.StartAPIRequestAndTimeGRPC("ExportNetworkHistory")()
+
+	if req.TableName == "" {
+		return formatE(ErrNetworkHistoryNoTableName, errors.New("empty table name"))
+	}
+
+	segments, err := t.networkHistoryService.ListAllHistorySegments()
+	if err != nil {
+		return formatE(ErrListAllNetworkHistorySegment, err)
+	}
+
+	ch, err := networkhistory.GetContiguousHistory(segments, req.FromBlock, req.ToBlock)
```

Comment:
This will fail if the passed heights do not sit on segment boundary, which is less that ideal, its solvable, but perhaps too much, but if so it needs to made clear that the heights passed must sit on segment boundary.  Else do something like-> 
  Ideally a user should be able to get any block which means some sort of interpolation on the network history segment index would be required  i.e for block at 1200, it would get the one immediately before  (span 1000->2000), then when the rows are being read out any with vega_time before that would be ignored.  Every history table has a vega_time column, but it would mean parsing on the fly and using the blocks file to get the relevent vega time, maybe this is too much?

Reply:
Yes it says in the docstring for the request
```
message ExportNetworkHistoryRequest {
    // Block to begin exporting from. Must be the first block a history segment,
    // which by default are 1000 blocks each - so 1, 1001 etc. are probably valid values.
    // you can check by first calling the API to list all network history segments.
    int64 from_block = 1;
    // Block to end exporting to (inclusive). Must be the last block of a history segment
    // which by default are 1000 blocks each - so 2000, 2000 etc. are probably valid values
    // you can check by first calling the API to list all network history segments.
    int64 to_block = 2;
```

But it is a little surprising. Before I noticed the function `GetContiguousHistory` I had actually written my own which just included the extra blocks before and after the requested window up to the initial/last segment boundary; but that's kind of surprising behaviour in it's own way, I can imagine users getting confused about duplicate rows. At least his way it's explicit.

I considered doing something like you suggested, but it makes things quite a bit more complicated. If there's a demand for it I'll do it (it sort of goes in hand with the time range comment above) - but I think for an initial cut what we have is a sufficient MVP.